### PR TITLE
LF-4155 Create form navigation buttons component

### PR DESCRIPTION
**Pull request by kathyavini**
**Description**

This PR includes:
- A new pure component `<FormNavigationButtons />` with customizable hint text. [Figma specification](https://www.figma.com/file/eMffC0h3JPlWtzNttxJpil/Litefarm-Updated?node-id=2722%3A51633&mode=dev). [Storybook Story](http://localhost:6006/?path=/docs/components-formnavigationbuttons--docs).
- A new wrapper component to handle floating (desktop) > full-width (mobile) positioning called `<FloatingContainer />`. [Storybook story](http://localhost:6006/?path=/story/components-floatingcontainer--default). There is also a storybook variant for the FormNavigationButtons where it has been placed within this container [here](http://localhost:6006/?path=/story/components-formnavigationbuttons--in-floating-container).
- A refactor of the `<ActionBar />` and `<ReleaseBadge />` components to use the floating container.
  - Note: if you want to test/view the ReleaseBadge, the easiest way is to set the initial state of `shouldShowBadge` in `<ReleaseBadgeHandler />` to true 

Jira link: https://lite-farm.atlassian.net/browse/LF-4155

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [x] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: packages/webapp/src/components/Form/FormNavigationButtons/index.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import styles from './styles.module.scss';
+import { ReactComponent as ChevronRight } from '../../../assets/images/buttons/chevron-right.svg';
+
+export interface FormNavigationButtonsProps {
+  isFinalStep?: boolean;
+  isDisabled?: boolean;
+  contextualContent?: React.ReactNode | string;
+  informationalText?: string;
+  onCancel?: () => void;
+  onContinue?: () => void;
```
Comment (SayakaOno) [inline packages/webapp/src/components/Form/FormNavigationButtons/index.tsx:29]: When are these not expected to be passed...?

File: packages/webapp/src/components/FloatingContainer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  distanceFromBottom,
+}: FloatingContainerProps) => {
+  return (
+    <div
+      className={clsx(
+        styles.container,
+        isCompactSideMenu ? styles.withCompactSideMenu : styles.withExpandedSideMenu,
+      )}
+      style={
+        distanceFromBottom && ({ '--bottom-offset': distanceFromBottom } as React.CSSProperties)
```
Comment (SayakaOno) [inline packages/webapp/src/components/FloatingContainer/index.tsx:39]: I didn't know that this was possible!

**Review by SayakaOno — COMMENTED** [review]
Wow, you created the FloatingContainer, and everything is nicely done. I love it! ❤️ 

One thing I noticed is that the width of the cancel button changes when it has a border, but other than that, great job!

https://github.com/LiteFarmOrg/LiteFarm/assets/33141219/f120b6e4-3492-4f68-a442-fba382463a6b

File: packages/webapp/src/components/Form/FormNavigationButtons/index.tsx (hunk 1 above)
Comment (kathyavini) [reply packages/webapp/src/components/Form/FormNavigationButtons/index.tsx:29]: Almost defeats the purpose of TypeScript there!!

**Comment by kathyavini** [comment]
Thanks for spotting that shift @SayakaOno. That looks to me like a general issue, so I edited Button itself rather than this view. Although it diverges from the CSS given in Figma, I think those focus state borders that expand the size of the button (not the large green one as it is a box-shadow) should really be outlines and not borders so they don't trigger layout shifts no matter where they are placed.

Could you take a look at the code adjust and see if that seems reasonable to you?

**Comment by SayakaOno** [comment]
@kathyavini Wise adjustments!
The focus state of the "secondary" button might be broken. In the story, the border is gone when it's focused.
The same thing happens to "secondary-2", but it looks like it has been like that.

**Comment by kathyavini** [comment]
@SayakaOno can you show me what you see? I think _previous_ to this last commit there was no border on the secondary focus state; I snuck that code change in too (alongside the outlines) when I noticed it. With the last commit it should be working now though:

Screenshot 2024-04-03 at 2 40 23 PM

**Comment by SayakaOno** [comment]
Oh my bad! I think I was seeing it with the previous commit!
It's all good :)

**Review by SayakaOno — APPROVED** [review]

**Pull request merged**